### PR TITLE
[MISC, FEATURE] Introduce global parallel config and add search_cfg::parallel

### LINKED ISSUE #1437: [Search] Add global parallel config to search interface
**Level 1 - Easy**

Similar to the debug config, the parallel config can be use for both alignments and search and possibly more. So it makes sense to make it a global config and make the algorithm specific implementations in the respective namespaces.

- [x]  Make a global parallel configuration that get's as template parameter a intergral_constant with the respective algorithm id type and id.
- [x] refactor the align_cfg::parallel configuration to use the global one
- [x] add a configuration for search in search_cfg namespace
- [x] add a test case for the parallel search config
- [x] document the search configuration

### PR DESCRIPTION
**Pull request by smehringer**
Resolves #1437

### CODE REVIEW

File: include/seqan3/alignment/configuration/align_config_parallel.hpp [hunk 1]
```diff
@@ -7,14 +7,13 @@
 
 /*!\file
  * \brief Provides seqan3::align_cfg::parallel configuration.
- * \author Lydia Buntrock <lydia.buntrock AT fu-berlin.de>
- * \author Jörg Winkler <j.winkler AT fu-berlin.de>
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
```
Comment (marehr) [inline include/seqan3/alignment/configuration/align_config_parallel.hpp]: you shouldn't change the authorship; you can add youself to that

File: include/seqan3/core/algorithm/configuration_element_parallel_mode.hpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
+
+namespace seqan3::detail
+{
+/*!\brief A global configuration type used to enabled debugging of algorithms.
```
Comment (marehr) [inline include/seqan3/core/algorithm/configuration_element_parallel_mode.hpp]: `to enabled debugging` what does that mean?

File: include/seqan3/core/algorithm/configuration_element_parallel_mode.hpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * \ingroup alignment_configuration
+ * \tparam wrapped_config_id_t The algorithm specific configuration id wrapped in a std::integral_constant.
+ *
+ * \details
+ *
+ * This type is used to enable the parallel mode of the algorithms.
+ */
+template <typename wrapped_config_id_t>
+struct parallel_mode : public pipeable_config_element<parallel_mode<wrapped_config_id_t>, uint32_t>
```
Comment (marehr) [inline include/seqan3/core/algorithm/configuration_element_parallel_mode.hpp]: What does the `uint32_t` mean at the end?

File: include/seqan3/search/configuration/detail.hpp [hunk 4]
```diff
@@ -39,14 +39,14 @@ namespace seqan3::detail
  */
 enum struct search_config_id : uint8_t
 {
-    //!\brief Identifier for max_errors configuration.
-    max_error,
-    max_error_rate,
-    output,
-    mode,
+    max_error, //!< Identifier for max_errors configuration.
+    max_error_rate, //!< Identifier for max_errorrate configuration.
```
Comment (marehr) [inline include/seqan3/search/configuration/detail.hpp]: `max_errorrate` ? is that a single word?

File: include/seqan3/alignment/configuration/align_config_parallel.hpp (hunk 1 above)
Comment (smehringer) [reply include/seqan3/alignment/configuration/align_config_parallel.hpp]: Alright. I just thought I literally changed every line of code so they wouldn't recognise their code and should not be held responsible for it. :innocent:

File: include/seqan3/core/algorithm/configuration_element_parallel_mode.hpp (hunk 2 above)
Comment (smehringer) [reply include/seqan3/core/algorithm/configuration_element_parallel_mode.hpp]: ops, copy and paste error from debug mode.
It is supposed to mean that the debug matrices are computed during the alignment.

File: include/seqan3/core/algorithm/configuration_element_parallel_mode.hpp (hunk 3 above)
Comment (smehringer) [reply include/seqan3/core/algorithm/configuration_element_parallel_mode.hpp]: The `pipeable_config_element` is a CRTP base that holds a single member variable called `value` that is of the type of the second template argument (in this case `uint32_t`). So it means that the configuration element will take one integer parameter (the number of threads, this is documented in the actual public configurations `align_cfg::parallel` and `search_cfg::parallel`)

File: include/seqan3/search/configuration/detail.hpp (hunk 4 above)
Comment (smehringer) [reply include/seqan3/search/configuration/detail.hpp]: :see_no_evil: of course not

File: include/seqan3/search/configuration/parallel.hpp [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+/*!\file
+ * \brief Provides seqan3::search_cfg::parallel configuration.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/search/configuration/detail.hpp>
+#include <seqan3/core/algorithm/configuration_element_parallel_mode.hpp>
```
Comment (rrahn) [inline include/seqan3/search/configuration/parallel.hpp]: alphabetical order

File: include/seqan3/search/configuration/all.hpp [hunk 6]
```diff
@@ -18,6 +18,7 @@
 #include <seqan3/search/configuration/max_error.hpp>
 #include <seqan3/search/configuration/max_error_rate.hpp>
 #include <seqan3/search/configuration/mode.hpp>
+#include <seqan3/search/configuration/parallel.hpp>
```
Comment (rrahn) [inline include/seqan3/search/configuration/all.hpp]: alphabetical order

File: test/snippet/search/configuration_parallel.cpp [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#include <seqan3/search/configuration/max_error.hpp>
+#include <seqan3/search/configuration/parallel.hpp>
+
+int main()
+{
+    // Enable parallel execution of the search algorithm with 8 threads (and allow 1 error of any type).
+    seqan3::configuration const cfg1 = seqan3::search_cfg::parallel{16} |
```
Comment (rrahn) [inline test/snippet/search/configuration_parallel.cpp]:
```suggestion
    seqan3::configuration const cfg1 = seqan3::search_cfg::parallel{8} |
```

**Pull request merged**